src/os/bluestore/BlueStore.cc

```diff
@@ -6399,7 +6399,11 @@ int BlueStore::statfs(struct store_statfs_t *buf)
     // part of our shared device is "free" according to BlueFS
     // Don't include bluestore_bluefs_min because that space can't
     // be used for any other purpose.
-    buf->available += bluefs->get_free(bluefs_shared_bdev) - cct->_conf->bluestore_bluefs_min;
+    int64_t shared_available = bluefs->get_free(bluefs_shared_bdev) -
+      cct->_conf->bluestore_bluefs_min;
```

Comment:
IMO (if I understand bluestore_bluefs_min semantics properly) generally that's improper use of bluestore_bluefs_min. Consider two cases (when bluefs_shared_bdev = 1Gb):
1) bluefs->get_total() = 2Gb and bluefs->get_free() = 0.5Gb. This should result in shared_available = 0.5Gb
2) bluefs->get_total() = 2Gb and bluefs->get_free() = 1.5Gb  => shared_available to be = 1Gb
So looks like one should take get_total() into account when adjusting using bluestore_bluefs_min.
I.e. something like
shared_available = min(get_free(), get_total() - bluestore_bluefs_min);